Add nightly dependency bump workflow

ezhang6811 opened the pull request:
*Issue #, if available:*

*Description of changes:*
Add a workflow nightly-build.yml that checks for newest versions of our OpenTelemetry dependencies, creates a PR if there are version bumps, and runs tests.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

thpierce commented on .github/workflows/nightly-build.yml:
```diff
@@ -0,0 +1,172 @@
+  workflow_dispatch:
+  push:
+    branches:
+      - zhaez/nightly-build
```
Remove

thpierce commented on .github/workflows/nightly-build.yml:
```diff
@@ -0,0 +1,172 @@
+    secrets: inherit
+    permissions:
+      id-token: write
+      contents: write
```
Why not read?

vastin replied:
Need to change it to `read` according to https://github.com/aws-observability/aws-otel-python-instrumentation/blob/main/.github/workflows/nightly-build.yml#L102

vastin commented:
In Example PR: https://github.com/aws-observability/aws-otel-dotnet-instrumentation/pull/287, why is it `Updated to latest available versions` instead of exact version of each package?

ezhang6811 commented on .github/workflows/nightly-build.yml:
```diff
@@ -0,0 +1,172 @@
+          echo "has_changes=true" >> $GITHUB_OUTPUT
+          
+          git add .
+          git commit -m "chore: update OpenTelemetry dependencies to ${{ steps.get_versions.outputs.otel_dotnet_core_version }}"
```
added instrumentation version

ezhang6811 commented on .github/workflows/nightly-build.yml:
```diff
@@ -0,0 +1,172 @@
+        else
+          echo "Creating new PR..."
+          gh pr create \
+            --title "Nightly dependency update: OpenTelemetry ${{ needs.update-dependencies.outputs.otel_dotnet_core_version }}" \
```
Added instrumentation version in title

ezhang6811 commented:
> In Example PR: #287, why is it Updated to latest available versions instead of exact version of each package?

removed this and replaced with Core and Instrumentation versions. the contrib packages are all independently versioned, so there's no unified version that all packages are updated to at the same time.
